vendor/github.com/coreos/go-oidc/oidc/provider.go

```diff
@@ -353,9 +353,6 @@ func (p ProviderConfig) Valid() error {
 	if !contains(p.IDTokenSigningAlgValues, "RS256") {
 		return errors.New("id_token_signing_alg_values_supported must include 'RS256'")
 	}
-	if contains(p.TokenEndpointAuthMethodsSupported, "none") {
```

Comment:
if I'm reading the rfc correctly, the "none" algorithm should only be able to be used when the token is obtained directly from the token endpoint. I don't see code guarding against a none algorithm in the token provided for validation